esphome/components/mcp4461/mcp4461.cpp

```diff
@@ -0,0 +1,688 @@
+#include "mcp4461.h"
+
+#include "esphome/core/helpers.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace mcp4461 {
+
+static const char *const TAG = "mcp4461";
+constexpr uint8_t EEPROM_WRITE_TIMEOUT_MS = 10;
+
+void Mcp4461Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up mcp4461 using address (0x%02" PRIX8 ")...", this->address_);
+  auto err = this->write(nullptr, 0);
+  if (err != i2c::ERROR_OK) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->mark_failed();
+    return;
+  }
+  this->begin_();
+}
+
+void Mcp4461Component::begin_() {
+  // save WP/WL status
+  this->update_write_protection_status_();
+  for (uint8_t i = 0; i < 8; i++) {
+    if (this->reg_[i].initial_value.has_value()) {
```

Comment:
```suggestion
    if (this->reg_[i].initial_value.has_value() && std::isfinite(this->reg_[i].initial_value.value())) {
```
If only codegen is setting initial value, then there is no need for this suggested change because it's guaranteed to be finite and within range.

Reply:
question - can I prevent usage of the function except for codegen? I thought it needs to be public for codegen and would then also be available automatically in a lamda? so I definitely need to check if the value is finite and in range ?